### PR TITLE
chore: bootstrap release-release for automatic changelog & version changes

### PR DESCRIPTION
**Pull request by JimmyLv**
Configuring release-please for multiple node packges by using manifest-releaser, https://github.com/googleapis/release-please/blob/main/docs/manifest-releaser.md

- added google-github-actions/release-please-action@v3
- using `node-workspace` plugin, to update all packages' version with dependency
- updated the  `"group-pull-request-title-pattern"` to `"chore: release v${version}"`

## Demo

Please see [chore: release v6.74.1 by github-actions\[bot\] · Pull Request #7212 · ParabolInc/parabol](https://github.com/ParabolInc/parabol/pull/7212)

Google Chrome-parabolCHANGELOG md at ee164c101e57e620fdcf5bbeb7e7170f0b7f4026 · ParabolIncparabol-004208-20220921

Temporarily configuring the `default-branch`(https://github.com/ParabolInc/parabol/pull/7209/commits/37b16a1dafe58a04a8733921a2a65e38fbd7924e) to `release-please/bootstrap/default`, creating the expected effect of draft PR at https://github.com/ParabolInc/parabol/pull/7212 as a demonstration.
After this PR merge, a new release-please PR will be created based on the latest changes to `master`.

One detail to note: in addition to CHNAGELOG in the root directory, a [CHNAGELOG.md](https://github.com/ParabolInc/parabol/pull/7212/files#diff-2429a8f921914f4c5e53adb2d920ad333cc06718368e993d166909c1b6b8ff14) file is also generated in each `packages/*` directory, but the update log only contains the contents of the submodule. From my research, there is no [release-please configuration](https://github.com/google-github-actions/release-please-action#configuration) found to remove at the moment.

## ACs

- [x] changelog entries do not have fix, chore prefix etc.
- [x] category/component of prefix is added in bold, i.e. `chore(DX): add release draft notes` is converted to `**DX**: add release draft notes`
- [ ] will update the [Release Playbook](https://www.notion.so/parabol/Release-Playbook-c3d1b146ea7c4f2fbf9501ae578ced25) after this PR merged to master

### CODE REVIEW

**Comment by JimmyLv** [comment]
Hey @Dschoordsch @rafaelromcar-parabol , I updated the Release Please workflow, but still need to discuss with you about the **overall** release process, for now (you can see the demo above): 
1. If the bot help to create a release PR with `base` branch being `master`, that means we don't need to do some manual version updates in [Release Playbook](https://www.notion.so/parabol/Release-Playbook-c3d1b146ea7c4f2fbf9501ae578ced25) 
2. but for the `staging` branch to release, is it after the PR is merged, or should happend before? 
3. this workflow doesn't support `-rc1` pre-release yet, but I think if full automation is needed here should require a little tweaking

**Comment by Dschoordsch** [comment]
> 2. but for the `staging` branch to release, is it after the PR is merged, or should happend before? Or should we merge the PR to `staging` and then merge it back to `master` after we release the `production` branch?

The bot should create a PR that targets staging with the release version and changelog entries prepared. Release owner reviews it, merges it and then staging is tested.
Once staging is tested, I suppose we would merge it to production and then production to master.

I wonder how the bot would do the tags in this case? This would only work for fast-forward merge, wouldn't it?

**Comment by rafaelromcar-parabol** [comment]
If it targets Staging, it would only create the PR and add the content once we have merged master into staging, isn't it? We will then not see the changes that are merged into master and waiting to be released.

I would say we could change our current release process to something like:
- Merge **production** into **master**
- Release Please adds whatever was missing and we merge the PR if we are ok.
- Merge **master** into **staging** (that deploys to staging) using `git checkout staging && git  merge vA.B.C` 
  - If something goes wrong, we make changes in **staging** until it works as fix(component) and we merge **staging** into master. That creates a new Release Please PR that generates a new version of the app (new patch in that case). We merge the PR and release the new version in **master**. We then merge **master** back to **staging** again, to release it again.
- Merge  into **production** using `git checkout production && git  merge vA.B.C` and that deploys to **production**.
  - If something goes wrong, hot-fixes in **production** that are sent back to **master** and a new version generated and we do the above process again, merging that new tag into **production** to release it.

I see a problem: we should not merge anything into **master** while the release is ongoing. But well, I think it's not a big deal to do that at the end. It does not stop developers from working, it just stops integrating their work into **master** for a while.

Just remember that once we would be into the Kubernetes logic, we will only have master and dev-branches. No prod and no staging. Master will be used to test things and build the docker image everything we merge the PR from Release Please (actually, every time there is a tag). Then, deployment will be done in a different place using a similar logic to the one I just described.

**Comment by rafaelromcar-parabol** [comment]
I was playing with release-please in the terraform module I'm building for the future GCP deployments and I found out it doesn't add to the Changelog commits starting with chore or anything else than fix and feat I think.

Also, I think that, if using release-please, we really should start using the scope in all our commits to master. It makes the changelog much easy to read that way. Here is [my current PR](https://github.com/ParabolInc/terraform-module-parabol-gcp/pull/2) by Release Please

**Comment by JimmyLv** [comment]
> I wonder how the bot would do the tags in this case? This would only work for fast-forward merge, wouldn't it?

@Dschoordsch When the Release PR is merged, release-please should automatically add the corresponding git tag. As the document said:

1. _Updates your changelog file (for example_ `_CHANGELOG.md_`_), along with other language-specific files (for example_ `_package.json_`_)._
2. _Tags the commit with the version number_
3. _Creates a GitHub Release based on the tag_

And that PR will go through these lifecycles in the process: `autorelease: pending` -> `autorelease: tagged` -> `autorelease: snapshot` -> `autorelease: published` (for this final step, release-please does not automatically add this tag, but we recommend it as a convention for publication tooling) [^1]

[^1] https://github.com/googleapis/release-please#whats-a-release-pr

**Comment by JimmyLv** [comment]
> If it targets Staging, it would only create the PR and add the content once we have merged master into staging, isn't it? 

@rafaelromcar-parabol yes, this could be interesting. I just got another very simple solution:
- Just change the Pull Request's `base` branch from `master` to `staging` and do the merge when releasing.

🤔 What if the branch created by the release PR is deployed directly to the `staging` environment each time? (i.e. as a notional staging branch):

- Still use `master` as the base branch 
- The branch of the PR created by the `release-please` is named `release-please-branches--master`, here's a real-world example: https://github.com/ conventional-changelog/standard-version/pull/867 
- Merging to `staging`
  - option 1: change the Pull Request's `base` branch from `master` to `staging`, then merge to `stating` which trigger the deployment on CI
  - option 2: 🤔 Currently `staging` branch is not normally used, if we allow deployments every time when a new update is pushed to the PR, we can also find any deployment issues in advance (e.g. the Nx issue in this v6.75.0 deployment, [Slack](https://parabol.slack.com/archives/C836NA350/p1663770560285449?thread_ts=1663745651.812859&cid=C836NA350))
- Each time we get to release day, Release Owner just needs to verify that the `staging` meets the test requirements
- If everything is OK, Release Owner can actually merge from the staging (`release-please-branches--master`) branch directly into the `production` branch
- Then finally merge the `production` branch back to `master`, and the changelog, version number and Git Tag are all ready to go.

Is it theoretically possible? I think I'll add a diagram of the branches later to make the flow clearer. 😁

**Comment by JimmyLv** [comment]
I've added the Git branching diagrams, hopefully it will help:

1. Before: Manually update CHANGELOG.md & versions
image

2. After: Automatically create the **release PR** to update CHANGELOG.md & versions
image

3. Final: Deploy PR branch before merge based on K8s, only have master and dev-branches (the release pr will also be deployed)
image

[view on diagrams.net](https://viewer.diagrams.net/?tags=%7B%7D&highlight=0000ff&edit=_blank&layers=1&nav=1&title=Git%20branching%20flow.xml#R7V1Zk6NGEv41ivA%2BSEFx8zh9zWzYHvf6CHv2ZYOWkISNhAxoptu%2FfqvEIagsoATFoRbjYwQSJUTeWZlfzpT73evHwD5sf%2FRXjjeTpdXrTHmYyTLC%2F%2BC%2FyJm3%2BIyBjPjEJnBXyYfOJ35x%2F3GSk1Jy9uiunLDwwcj3vcg9FE8u%2Ff3eWUaFc3YQ%2BN%2BKH1v7XvFbD%2FbGASd%2BWdoePPu7u4q2yVkkSec3PjnuZpt8taklb%2Bzs9MPJiXBrr%2FxvuVPK40y5D3w%2Fil%2FtXu8djzy89LnE1z2VvJvdWODsI54Ljr%2F%2Ftv%2F8%2BKB%2FevRk%2B3ttvbL%2F%2FTJPiPHV9o7JD%2F7kR2v3FZ%2B793c7N5rJuoeXv3sJ8KsNebVYLODJveOsCJWdg%2Be%2FkUcfEkJtHfK77Z3DucoycOyIXPMS2PvlltAr8HdkjcjeuPtN7p3TE43eUjJFzitZYBvtPHwC4ZdhFPh%2FOfe%2B5wf4zN7f40%2FerV3Po07ZnrvZ48MlfowOPn%2F31QkiFzPAh%2BSNnbtaka%2B5%2B7Z1I%2BeXg70k3%2FkNszs%2B5%2BNPr70TWbf4cw6%2B4A4SJqEVWdl5zZ1KCPXR8XdOFJDnlryrqfLCNOOrEsFBhh4ff8uxYcpb2xwHIiU5aSesv8mWP3MHfpEwyAXMYgJmAXRwVlh6kkM%2FiLb%2Bxt%2Fb3uP57F3gH%2Fcrh3yNVCTYn04UvSUKwD5GPnm62Qo%2F%2BP4h%2BdzBs939%2FOV0D3fOfvWBSPmZnvjMk0t%2B12n9mAtSyVUBX8xkRTr9OV2JH9IfyYWngy%2FkYKGlhw%2Bv%2BTcf3vJHz07g4qdMOOh0spQJQv8YLJ2Kp5xqvsgONk5yrbW9%2B%2F754R%2Ffjb4d%2Fn23%2Bufp4T%2FGHCUMQZ45D1PJlop%2FS4GndFTOKsk6z76Lf0G2iKrWLhLfeHIdxXLZXTXnQqsB0y2PwdcTz6EqDnRe3eiP3Osc%2BfERTeFadoEXVPMqrwBAnl77%2Byi5EGnCmY%2F9QY2T%2BQLHsyP3a9GclvIbfkD2W%2B4DB8JFYTk7Kqmyo5jxifPzsqpTPBrfgVCOTR9qleJMrc4P9ovjPfuhG7k%2BsT4vfhRhEwjNUkT4ARq8gqrb2gey%2FO51Q5yzBTFUy60dRAt%2FPyeOy%2F8CZ%2B1gJiEsUdSr%2BIKVi7mnwn5i1XmnPSpPTqVSFWAMZUumVY4KbaHCMIXdWUIOirbSSkzBZyiIvPa62BSOxeChTlQJkH1TVwY2XekvvX5FUC7wLBXxKEYRmBJayCPTBGl8O7BTzBLuOnWA6U99opRG9e6AxJThovS%2F%2FWr%2F%2FPz5e9fT%2F%2Fpto375w%2F9hf8AyIVcSHYnwWpEmA86hV4nvtDvZVwCj7OyQ6M8xx7J5zzLzNJNbRcnxk71zPfJU7%2FETdMkPkj4738SIvIUocZeAuGsMcVfl9uJewq2WcMvfwNbnySKLtuSX65KVHW6zn0MOnu0Ir7c%2FnZEllPECUA%2BN%2FAUNKiU2sZSWfkQ7y6Bdg7EHhnylv%2BiaDg35er2Wl0sxUq2k8plGYQoa1orL0J%2FPUo7CtbPnrKN3qJtlBbpissqgotwVFQ0O97peN%2FcSehUMQHNVqFtMvSo8dNIUhv9EkVFc7MRU5jLMPu8c%2FI3EEPgzst8SO1OSvY7%2Fxj%2FZsUOHyQEnfVukIb%2FvFDih%2B4%2F9clqPUDfJUeHFtbuZ9sAwzyWSle1aJYvNMtHI80IFq5eK5lxaIE0xKPmMD1uygqIWVp1rxQX89Tp0OnGgDa2BMAvIqiSeVXbQwrMaXE9cTVrXKqZpFVPLs1Tt53tJ61pqA4YUnfMT6Pdfzp2VHgZ5s5AgsqzqjHDXRkxGjZMAkOFALlHVKU1YYg8vFYWsgiJlbUOqvjN6h%2BPCz6e%2Fo1PR0Tmi6OFjozgRejgGB09EKvRBezRPd1EqAfy%2Bt5pa81RBGjAx0msElVYn5Ch6CPzVcXmi2hRElQZRKkVHBdAxjbN6iaFMITHUAJtZF%2BWza7PSvP6SqYq2Xa2oZxnDuySVRRRnf4QcvOUObs0b0RWT9kaAC8HrjRgcno0gd0Q3K92F2s9TBRqt3Qum9EJb5B%2Bwtse%2FEv%2F3fDwt8LPz99EJI6a8dJweKLERLdIBsdxXpgMUpegxIDHpgDmSi2kGJOnFNbrLCKTKd9xeJPAPLd1QbEaG3UErzTHEuBW6BLWLag3rIppQLNeOjTnjSdFUeT53Xu0d9rTn5OQxgIm7yWnMCiGKTqOqqYCy%2FWbeTVhH%2Fuup%2BDsfA3RXu90HXaUmTsQFcUAxylcsGAf0TFKYb%2F%2BCORmfyeQzIWh4quWfiMqQU%2BCWqcbgogpTL7GoZvslk5xWkdQq0FPWht70tODWdUzPne3mlS7dazORl6mGJbrYjJFYQ6q%2BSHej%2BqExT1LmHC8vPTsM3WWVG1vrpZQ944vSLJXVgGztJLPJU1PvlZ5ru7cpKTAmzxL9l0blqsZYTS2u1XWvjDwaxuFlkqFojywqo24qzcieXVi2UNc0h4WmE83L9tYoa67qzWguWzULdVwcgyTo1blrfEz6clnuef8lqiUWoVmTWsLi9RWgKpu9eqoAlaAo%2FmpvZsoHcg77EOpCmgdLBKhT54phGdNPf2ASaa2Rfy5KWLZxy3LfHP8R44bJKkxoW4z%2BZ6RmH%2BvFEUMIJjpGp1zz8rP%2BZf9f7ev%2By6%2Bhd699toPvd8H%2F5rw1Pv0oYUMyMm86obV%2BbjK%2BVBNrsNNDNSh2EKeMqx5vgUXi6Gvlfi2wiv730Y9jsdyrGAnBX8Ux%2BVcXMwHhZ93eEcGM%2F4%2FPrNxwecTM5e9D5vvHw8qOnDD9avxLTt9%2BVcGfzBP8iWs0VVUN8I7FyNdgrcMK%2FxS9nPV5tQ6ToYS0nLTe2GURphpGgTzMfFnarKYoLdk5Pu8Wf5md95HZO8fNlaAJXYhqI955eZoMGt3orRhRJWUq9T1mdYmYnBbWsz%2Ffeg%2B3ij6ttOiP9v5oe95bpg7xi%2FtPHz5%2FfPzhp4%2BL3apMOcZnw4O9Z34R4fx5eBIF4skhdHhlfv3lNwzVONZeiY6vulV8Nr7bhur9EoeyC0Wf6ZdyRS%2FSvUQm5W8wcnwsaB265UGYhmc0hn8k2EvS3Sl7G0MfPbj2JrB3YQfGu1mQUDQpOXqe6NShoZaNs8uYd1IVE9JMvZxmv4VO8NPLnwROTJa8uBYkUUCxXO5fQvLXh3XWvuu5%2B7%2FiD22jiECTfSCLy0977L8tlgTD6ungh0QpP9nz8LhcOmG4PnrzjRvNX1Iaz3cxfNpTvGbKMUzwHz33gBIOgxzBIhpt0alt61QiDagJVppjrlSW7jDlF%2BWkO0TIpqSCcECG8skKSBpULuDDHLXZEst8%2FjJPSr5fpyx2tsYPAlIJZlXvj%2FVUn6fpAHWmO7SQykcy7vKmSrQQrBseP%2Bh3J90AtMadqakao07%2Bybx%2FvL8Xo000Ux0SMoSdlARkpZKClycEh4SEKCEJg3DlVNKK6XPEUPim0dmeK5NKNalbbSHdGpUMQ14o5pCEqtgyHruaHKQKlNGl1mMRKFOseDBEgQ%2FTVfKcZzeqvgVEZtNgKlloxCBX0RnYVJpb6V9V0cYmzTzIlqPsFuOXf86SJY7NaItN3bYKwIKdyJTE8ko%2Fq2mgZ%2BlnIiQmpYt00pPo9hlPftYoyc%2FeOWs%2FcHI51HhFkFp9uSCvKrJPiDB6mPhwLKWzMqwXVqPxeu3oOaimVkqHat%2FSGP5erwqHo7hpePPQeUws2I70ib9VUadTVygDSf0eCmXa0VI2FzrV2NdzoQwbrQkGZTRBJ2ICYmoWnfpm9QH1T0yORGSvRU%2F1fttQ4ZiJtEUKhUHj4l3qjjHWQma%2FdaUaDNen%2BTBi5sO0S5jqxgIV1b7BcM26mw%2FD5pYOCpXe%2BYCYSyNQNtwBI1XFjkBLUlVpKIlM2v4YilbOLCU5JL1%2BlY7Vls6KIq8rSmg5cKOVdlGQTgUIBiPP32uAoEN%2FsoFqeYcDN8QokLZA2SUOjAkqu3vXBNefL2g5caNd9kemUwXDawIO0JdRDtzgl9daO67XlMcJsePIAFaAXkTcyAz2z4SdFLcxMqOVzCIEuhiMNL821NgM%2FSpKmZrOQmjnbOk6TS5THzgdq8MutwQdZH6I%2F0qKVp1wPm8okvmy0bJC0xHiNrVL8BU3UwwT2lL5rHV7ye3pQoa6DjUzQYj3a5RExW0LSDVoP%2FvdTGWgLN%2FGyIQKTi%2BVzSsfmcDmbLjJcpMjEwSqCR7vXGczmnDQAfZtwurhaSrB2YpfjmskhHlQPFhDuJHB7mPDII2BWwOyNUhSLC6LdXEHqg52OtP7Lm1ChZeYqlr9izTQStTgkuwhdApZnGqXcUdI3U5EaOdbqwYdR1kqDHt7jaMMmH28laEILVMYoBDRUg1Ay64GI7Bp2UHeceDBCEIsWzpyoNYtMkowyntyi5pMD5smIzQqtWk7GQF2XgI%2FhHsyQr1%2FJMilMSTYaUh5DjzXUG5QR87Gex6QUCH%2BlcH%2FdQ9IqFC5Y3cqB2mm0RWoZGRYa9mvuwhzsbc7IKGdAwk7WTB5TUDezqC62fQtg16%2FpSkJLQMDsFlm6TAw6Jes6S7OjU9KaCmwtJ%2BGJHVoeWXMwiuMSpgfglk5wP537n7pHVcEgvEMOjZjQTWmGF%2F%2FmniDY9fUlBi7piy%2BqCp5accXMFd32yMXWup0kHI1GIm7LocusKk8Hhz1rlM4JcFRTwgGsgHDfaMxgoHBWM0srtXxtjvPWLqpXSoWXgST9prRkPJS%2FVqdw%2FDzwINfCfQ%2Bm7cZMG0lj0JiM1Y%2F6V7GrMHl1g%2Bc7%2BKal3%2FFnajnkVctelK5q9hGDe6kWNIibbJIrTBr9FFJ72kDlE1OQo4NcF%2BMGJXUIvWEuK8UCY1IwXhTvH1QJY4Dt37L3BiTCifA%2Fb4A99tFdroFmEdnRP0dAu6XoIw02S1sVSm1sCRjVqiWkoxG9VJsZZcnMFNdnrcQ4V6nQMWXTiusL6Jru1vcrC7JUEAsqtWA42PDWX1JN3t4jLmPl2u9D5gjd%2FZJf5xQ8rMG%2B7Nn9PzzLCkXniD0RXmCfUHoC1DRMgISofMiDYiAy2fz%2FgS%2B3W1DcDo3sl5Nd9M5rJn60J3DPKMzh987HzfSGAEUHFn7MGM65s2jb5sA9FVnjL3qG4DbgmnUm0fgtnQa7W0UlGJNvBudqhykzIjZVDF0nZF1GZ7byEC4py2sHpyfqXCwNGcFQRpHINFCurivBoe70grV6wohqSUotojVP0fJLTcUN6M8tWcdgCQhTDXizUwk8WYjkdTXIPHqGxUMp%2Fs%2Bti5VGcQvpj4CDOSEaSaqsalmADvKKuUdgGoc0czIdpyH8ntNVVpISMr%2ByEUhRBSNuIGPK5c16LRy5yMprj8TOCSaaIZHP5r5ezxDJN47hGDC1RVUA9PI6fGXHBCCmgzacOhVeJWCrNfeUOeK4CryXKNVBPh90O069HYA4hmyOuHn1LsqaXN7%2Brppczu%2FiuPF3OFShjeKziNeRRpipYcbPgGyfVZZpRWkqKmcNLfqsEygXIQFWvjWMGQljKjCdp3GTK0D%2FAOTDuwEwTMo2BW5EHGKcUnae1Yupajmkm6KwbKZHgyox29uRCalkDouhmRePaxjonJKvQ5pYagGhcAwExH8pnO8e0d1ROm95oidDcC5MJ00jg7DXiABMCdwZAl7xZRCqbNZ1mE8m%2BAAKmkKccJME9K0365xJI%2BncZTXnRgqgSgjCxDQaDguTYEbs2CtrjMEMseW2VgyBKNEcSTeJeWlGXBToN8UgQJTBHuifcvxsgccC1yu97ldsQvq0QyQ0WGNkrX6JResXaHJdXKGaXQG2wsce3XuglgR40sO4zGFWDzHRGeefcAuKI6VbFqHUIHHgXQGyTvrEUXp3JL3OniQ14zzJMXS5sz6pJhakkBIVTUCZWodOlWq0oCi3SaWSpKzwyWWhmETbVC4%2B%2BxGr8HlGuOmjAmlOK2AGG4aPE8JxuinPdZGbKns6qJllxukQ7WGld2rGAvWbjqjABnVYEH4CGRUPNTNTcmoxW1f26Jst6Sz%2BCrgsdN5rF25FzCNGPgEBmA5KEtXELXx0XWOLX0Ik9GoNBq6RcGZD28yNLhJ%2Bd5ViUjp17hr%2BzR5UJOR3ugkoZUSaqQQAyOSUI5E2kQ52TThrI8REG8qa07TQeXpjguTliV1zQZtW%2Bk1Op6MjjSY9ZpGo9fKLZJouiG4d8Ha7u4ur60Jnks15bVrnCju3JhWgtHZlxN1Rbmx5qUEAoRaSfcas4iUsSHZsyUeXxX0%2BxZqk1eo9UFRqbMbnYS6rjuYhigegVBbk1D3KdSpWm9eqNCTUOscaa1JqE%2BeNZ07HlyodZiqmkbuNYutEHDDYEKk50rsdKDBNKhJTPQMkK5RJpr54j9F7hmyQ4eZr5IJfBNly8Zl0pRVGJSVpd4p2wGEzpWDgMqqQYOAImQyxBATNVWAPRELlm3dOmKrroGGUDaxULax3hOtRol7cJtlHSnsVQ8bu%2Bz9DAtpYD8r8837quvQrwJzdNyA63BL2Rx6V1KHKa9PfrR2X%2FG5e3%2B3c3naaoazCSKIYjCKpqAJMPslC6yiomz1zeHgM3qbUDbDpNDO1rMTbMAEU4FW%2Bu35VSb2q0yKViy%2FymJPDuuQVjCh5B8ckmyw8X%2FPx9MSPzt%2FH50QKr4eADlKnz0PAEdNzt5UQG2F2d2DhrmdB%2Bfg%2BW8covDeuzRVk567gsN56AiwmjQbWJzfQif46eVPzDz4Oi9mX3Lpk4ud%2F3g1z93%2FFZ%2FcRtEhPKmuJ%2Fzv%2FqvrLJbYc5OfDn5I8Hue7Hl4XC6dMFwfvfnGjeZx4sbdb%2BY7f4XXJhfK9eyRH0yUMBTkCZYGo7GnqRxqSiqjgqTlTCCAuKCIBTFsFKuIpQFp8WGOuheJ52VDsBPbs7LDbRZP5ghRDC3rRxzOOKMz%2BJD7wb1I0%2BAJCTVaTfJiXlhyzUJdB20GT6fezZLZAGoY0ULITWmtfq3GxMaHge9H%2BY%2BTUPjHk75VHv8P) (previously draw.io)

**Comment by Dschoordsch** [comment]
@JimmyLv I'm not sure I get the full picture. How is this whole process triggered?
The flow of a bot branching master with changelog and version changes seems sensible.
To my understanding the flow is
1) somehow the bot is triggered, it creates a PR with CHANGELOG.md and version changes targeting staging
2) release owner checks and merges the PR
3) after the PR was merged, the bot needs to come in and tag the commit
4) tests are done on staging
5) staging is merged to production ff-only
6) production is merged to master

Would this bot do 1) and 3)?

@rafaelromcar-parabol I'm strongly opposed to having a merge freeze to master, that wouldn't be a step forward.

File: packages/client/utils/draftjs/entitizeText.ts [hunk 1]
```diff
@@ -13,8 +13,8 @@ const entitizeText = (contentState: ContentState, selectionState: SelectionState
     const currentStart = anchorKey === currentKey ? anchorOffset : 0
     const currentEnd = focusKey === currentKey ? focusOffset : currentBlock.getLength()
     const blockText = currentBlock.getText().slice(currentStart, currentEnd)
-    for (let i = 0; i < textTags.length; i++) {
-      const tag = textTags[i]!
+    textTags.forEach((item) => {
+      const tag = item!
```
Comment (Dschoordsch) [inline packages/client/utils/draftjs/entitizeText.ts:17]: -1 `item` is of type `string` already and the whole change seems unrelated

**Review by Dschoordsch — CHANGES_REQUESTED** [review]

**Comment by JimmyLv** [comment]
> Would this bot do 1) and 3)?

@Dschoordsch yeah, your understanding is exactly correct! the bot will _[tags the commit with the version number](https://github.com/googleapis/release-please#whats-a-release-pr)_. 

> 1) somehow the bot is triggered, it creates a PR with CHANGELOG.md and version changes targeting staging
release owner checks and merges the PR
> 2) release owner checks and merges the PR

For 1) and 2), I would like to add: currently, the Bot only can monitor one branch and create a PR for the corresponding branch, so the Release Owner needs to manually [change the PR base](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-base-branch-of-a-pull-request) from `master` to `staging` and then merge it, thus this PR actually includes these 2 steps (previous manual):

1. merge the `master` Git changes to the `staging` branch 
2. change the package version number on the `staging` branch

How does that sound?

**Comment by mattkrick** [comment]
Just a question about hotfixes-- today we branch off of production & merge the hotfix back to production. This is because master (and staging, technically) could have additional PRs that we don't want to introduce at the same time as the hotfix. Would this be the same going forward?

**Comment by Dschoordsch** [comment]
> > Would this bot do 1) and 3)?
> 
> @Dschoordsch yeah, your understanding is exactly correct! the bot will _[tags the commit with the version number](https://github.com/googleapis/release-please#whats-a-release-pr)_.

You linked the documentation before, but it lists the steps in the wrong order, that's why I was asking. When we merge the PR, we will do a squash+rebase, i.e. the hash changes and the tag is no longer valid. Does it update the tag after the PR was merged?
 
> > 1. somehow the bot is triggered, it creates a PR with CHANGELOG.md and version changes targeting staging
> >    release owner checks and merges the PR
> > 2. release owner checks and merges the PR
> 
> For 1) and 2), I would like to add: currently, the Bot only can monitor one branch and create a PR for the corresponding branch, so the Release Owner needs to manually [change the PR base](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-base-branch-of-a-pull-request) from `master` to `staging` and then merge it, thus this PR actually includes these 2 steps (previous manual):
> 
> 1. merge the `master` Git changes to the `staging` branch
> 2. change the package version number on the `staging` branch
> 
> How does that sound?

Sounds ok from a workflow perspective.

**Comment by JimmyLv** [comment]
@mattkrick yeah, hotfix is the case. I found release please [support multiple release branches](https://github.com/google-github-actions/release-please-action#supporting-multiple-release-branches), so it can monitor `master` and `production` branches by configuring multiple workflows that specify a different `default-branch`:

Configuration for `production` branch (`.github/workflows/release-producation.yaml`):

```yml
on:
  push:
    branches:
      - production
name: release-please
jobs:
  release-please:
    steps:
      - uses: google-github-actions/release-please-action@v3
        with:
          default-branch: production
```

So when branching off of production & merge the hotfix back to production, the release-please bot will automatically create the release PR to automatically modify the versions and CHANGELOG, and of course eventually the production branch will merge to master.

**Comment by JimmyLv** [comment]
Hey @Dschoordsch , you're correct 😁 it lists the steps in the wrong order. So I tried it out myself, and after this Release PR (using the `staging_2_test_and_will_delete` branch) merged, the corresponding Git Tags were created successfully. (will delete them later)

By the way, I found that the git tags contained package component names, so I changed the `include-component-in-tag` configuration [^1]. Could you please revisit the workflows config file for this PR?

Google Chrome-chore release v6 75 1 by github-actions bot  · Pull Request #7229 · ParabolIncparabol-000479-20220928

[^1]: https://github.com/googleapis/release-please/blob/main/schemas/config.json#L77

File: .release-please-manifest.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  ".": "6.75.0",
```
Comment (Dschoordsch) [inline .release-please-manifest.json:2]: +1 I just did the release and this should now be 6.76.0

**Review by Dschoordsch — APPROVED** [review]
Let's try this. I propose when we merge this now, you should do the next release and update the Release Playbook to the new workflow.

**Pull request merged**